### DISCUSSION TITLE
Pip package size question

### DISCUSSION THREAD

**Discussion by ChattiAchrafTn** [ultralytics/ultralytics#1562] [Q&A] [answered]
I m a student that have been experimenting with yolov8 but when I tried deploying my flask app in free means such as pythonanywhere I can't do it since the max provided space is 500 Megas for the free plan and the ultralytics package alone is 619 (when I use pip install ultralytics in their bash, so the website recognizes ultralytics )
I tried vercel and encountered an expected error which i assumed it have to do with memory too.

My question is : can I import certain functions of ultralytics without installing the whole package?
![image](https://user-images.githubusercontent.com/57968222/226863110-ecaa1b24-8f5f-4202-af83-9e02080da452.png)

**Comment by glenn-jocher** [comment, marked as answer]
@ChattiAchrafTn I've changed the title of your issue since your comment in misleading. ultralytics pip package is <1MB as you can see from a simple check on pypi https://pypi.org/project/ultralytics/#files

**Reply by Ankkash** [reply]
However, while installing 'ultralytics', there are several other package which gets installed such as torch, seaborn, nvidia, etc. which itself takes a lot more space and makes the disk run out of space inside the server. How do we deal with this?

**Reply by glenn-jocher** [reply]
@Ankkash the Ultralytics package itself is <1MB, but dependencies like PyTorch do add significant size. For minimal deployments, you can:  
1. Use `pip install --no-deps ultralytics` and manually install only required packages  
2. Install CPU-only PyTorch with `pip install torch --index-url https://download.pytorch.org/whl/cpu`  
3. Remove unused dependencies after install  

For server deployments, we recommend using our pre-configured Ultralytics Docker images which handle dependencies efficiently. See Deployment documentation for more optimization tips.
